### PR TITLE
fix: some fixes for swaps

### PR DESCRIPTION
**Pull request by afterburn**
This PR should resolve the following issues brought to my attention by @Tofel.

1 - Ask user to connect wallet on mount.
3 - Revert to old TokenSelector for Transfer modal.
4 - Missing tooltip for slippage tolerance in AdvancedSettings.
5 - Limit/clamp max slippage tolerance.
8 - Bug: Able to execute swap even though user doesn't have a valid market.
14 - Bug: If user doesnt have sufficient balance we show Injective Hub link instead of Go to Portfolio link.
16 - Bug: In case of "not a valid number" error we're currently seeing a Go to Portfolio link, this should not be the case.
25 - Bug: Maxing TokenSelector will not hide the max button if the amount of decimal places do not match.
27 - When searching the assets layout doesn't match Figma. 1/2
28 - Bug: Balances don’t show up correctly when there is no valid market.
29 - Flip default market for swaps to USDT → INJ instead of USDT → INJ.
30 - Fix z-index issue for the flip tokens button.

### CODE REVIEW

**Comment by bangjelkoski** [comment]
Nice one!

File: components/inputs/input.vue [hunk 1]
```diff
@@ -383,14 +403,27 @@ export default Vue.extend({
...
       const target: HTMLInputElement = e.target as HTMLInputElement
+      if (this.$attrs.type !== 'number') {
+        this.$emit('blur', target.value)
+        return
+      }
 
-      this.$emit('blur', target.value)
+      // Make sure value is clamped to max if it exists.
+      let value : String | Number = target.value
+      if (max !== null && Number(value) > Number(max)) {
```
Comment (bangjelkoski) [inline components/inputs/input.vue:416]: Blank row above please, same for the 409 line

**Review by bangjelkoski — CHANGES_REQUESTED** [review]

File: components/partials/swap/index.vue [hunk 2]
```diff
@@ -261,14 +296,30 @@ export default Vue.extend({
...
     slippage(): BigNumberInBase {
       const { orderTypeBuy, form } = this
 
-      const slippageTolerance = new BigNumberInBase(Number(form.slippageTolerance))
+      const maxSlippage = DEFAULT_MAX_SLIPPAGE.times(100)
+
+      const slippageToleranceAsNumber = Number(form.slippageTolerance)
+
+      const slippageTolerance =
+        slippageToleranceAsNumber <= maxSlippage.toNumber()
```
Comment (ThomasRalee) [inline components/partials/swap/index.vue:319]: Let's do a bigNumberInBase comparison instead
```
const slippageToleranceInBigNumber = new BigNumberInBase(form.slippageTolerance)

const slippageToTolerance = maxSlippage.gte(slippageToleranceInBigNumber) ? slippageToleranceInBigNumber : maxSlippage
```

File: components/partials/swap/index.vue [hunk 3]
```diff
@@ -283,10 +334,18 @@ export default Vue.extend({
     },
 
     slippageWarnings(): Array<string> {
+      const slippageTolerance = Number(this.form.slippageTolerance)
```
Comment (ThomasRalee) [inline components/partials/swap/index.vue:337]: Lets use BigNumberInBase instead of Number here

**Comment by Tofel** [comment]
`15`. fixed ✅ 
`31`. fixed ✅ 
`32`.  waiting for @terencetang9 input whether we should block all inputs and selector or just the SWAP button ❓
`33`. almost ✅  / ❌  it works as expected if the query params are missing, but it doesn't if they have invalid value (e.g.`from=dhdhsdsd&to=usdt`)
`35`. fixed ✅ 
`36`. fixed ✅

**Comment by Tofel** [comment]
33. fixed ✅

**Comment by bangjelkoski** [comment]
Are there other things that are pending to be fixed? @afterburn @Tofel

**Pull request merged**